src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc

```diff
@@ -1,23 +1,251 @@
+  components:
+    - alias: mysql
+      type: kubernetes
+      reference: petclinic.yaml
+      referenceContent: |
+           kind: List
+           items:
+            -
+             apiVersion: v1
+             kind: Pod
+             metadata:
+              name: ws
+             spec:
+              containers:
+              ... etc
+----
+
+As with the xref:component-type-dockerimage_{context}[`dockerimage` component], it is possible to override the entrypoint of the containers contained in the Kubernetes or OpenShift list using the `command` and `args` properties (as link:https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes[understood] by Kubernetes).
+
+There can be more containers in the list (contained in pods or pod templates of deployments). To select which containers to apply the entrypoint changes to.
+
+The entrypoints can be defined as follows:
+
+[source,yaml]
+----
+  components:
+    - alias: appDeployment
+      type: kubernetes
+      reference: app-deployment.yaml
+      entrypoints:
+      - parentName: mysqlServer
+        command: ['sleep']
+        args: ['infinity']
+      - parentSelector:
+          app: prometheus
+        args: ['-f', '/opt/app/prometheus-config.yaml']
+----
+
+The `entrypoints` list contains constraints for picking the containers along with the `command` and `args` parameters to apply to them. In the example above, the constraint is `parentName: mysqlServer`. which will cause the command to be applied to all containers defined in any parent object called `mysqlServer`. The parent object is assumed to be a top level object in the list defined in the referenced file, which is `app-deployment.yaml` in the example above.
+
+Other types of constraints (and their combinations) are possible:
+
+`containerName`:: the name of the container
+`parentName`:: the name of the parent object that (indirectly) contains the containers to override
+`parentSelector`:: the set of labels the parent object needs to have
+
+A combination of these constraints can be used to precisely locate the containers inside the referenced Kubernetes list.
+
+
+[id="component-type-dockerimage_{context}"]
+=== Component type: dockerimage
+
+A component type that allows to define a container image-based configuration of a container in a workspace. A devfile can only contain one component of the `dockerimage` type.
+
+[source,yaml]
+----
+ components:
+   - alias: maven
+     type: dockerimage
+     image: eclipe/maven-jdk8:latest
+     volumes:
+       - name: mavenrepo
+         containerPath: /root/.m2
+     env:
+       - name: ENV_VAR
+         value: value
+     endpoints:
+       - name: maven-server
+         port: 3101
+         attributes:
+           protocol: http
+           secure: 'true'
+           public: 'true'
+           discoverable: 'false'
+     memoryLimit: 1536M
+     command: ['tail']
+     args: ['-f', '/dev/null']
+----
+
+
+== Adding commands to a devfile
```

Comment:
This will need to be expanded. We also support embedded editor command definitions (e.g. tasks.json and launch.json).

Reply:
PR, please? :)

Reply:
Just an issue, for now :) https://github.com/eclipse/che-docs/issues/734